int32 id
string skill_name
string step_name
string param